src/components/cluster/detail/ScaleClusterModal.js

```diff
@@ -415,18 +418,31 @@ ScaleClusterModal.propTypes = {
   cluster: PropTypes.object,
   clusterActions: PropTypes.object,
   provider: PropTypes.string,
+  maxWorkersPerClusters: PropTypes.number,
   workerNodesRunning: PropTypes.number,
   workerNodesDesired: PropTypes.number,
 };
 
+function mapStateToProps(state) {
+  let maxWorkersPerClusters = 0;
+
+  if (state.app.info.workers.count_per_cluster.max) {
+    maxWorkersPerClusters = state.app.info.workers.count_per_cluster.max;
+  } else {
+    maxWorkersPerClusters = state.app.info.workers.count_per_cluster.default;
+  }
```

Comment:
I don't think it makes sense to use the `workers.count_per_cluster.default` value (normally `3`) as the maximum.